zrml/prediction-markets/src/tests.rs

```diff
@@ -104,19 +104,25 @@ fn simple_create_scalar_market(
 }
 
 #[test]
-fn admin_move_market_to_closed_successfully_closes_market() {
+fn admin_move_market_to_closed_successfully_closes_market_and_sets_end() {
     ExtBuilder::default().build().execute_with(|| {
-        frame_system::Pallet::<Runtime>::set_block_number(1);
+        run_blocks(7);
+        let now = frame_system::Pallet::<Runtime>::block_number();
+        let end = 42;
         simple_create_categorical_market(
             Asset::Ztg,
             MarketCreation::Permissionless,
-            0..2,
+            now..end,
             ScoringRule::CPMM,
         );
+        run_blocks(3);
         let market_id = 0;
         assert_ok!(PredictionMarkets::admin_move_market_to_closed(Origin::signed(SUDO), market_id));
         let market = MarketCommons::market(&market_id).unwrap();
         assert_eq!(market.status, MarketStatus::Closed);
+        let new_end = now + 3;
```

Comment:
I think this should be (probably missing a cast here :wink:):

```suggestion
        let new_end = now + 3 * MILLISECS_PER_BLOCK;
```

Reply:
We are using `MarketPeriod::Block` here. If we had used `MarketPeriod::Timestamp` here I would totally agree.

Reply:
Oh, my bad.

Reply:
So there's no test for this when using timestamps? If not, could you add one? Seems important.

Reply:
ad3f275465481a1d7bd95990f883c5b550bac3d9